### ISSUE TITLE
Rancher project member with read-only cannot view monitoring graphs at bottom of cluster page

### ISSUE THREAD

**Issue by ronhorton** [rancher/dashboard#9792]
**Setup**
- Rancher version: 2.7.7-rc7
- Rancher UI Extensions: N/A
- Browser type & version: Chrome Version 116.0.5845.187 (Official Build) (x86_64)

**Describe the bug**
When a standard user is added as a read-only project member, the user does not have the grafana monitoring graphs at the bottom of the cluster page. 

**To Reproduce**
1. Enable monitoring v2 on downstream cluster
2. Create a project in this downstream cluster
3. Create a namespace in this project
4. Create an nginx workload in this namespace
5. Open grafana as admin user (from monitoring page) and save the URL for later use
6. Edit the members of the project and add a user with the "Read-only" role
7. Run this in the kubectl shell for the cluster:
8. kubectl create clusterrolebinding view-monitoring-ui --clusterrole=monitoring-ui-view --user=userID
9. Sign in as new user and navigate to downstream cluster page

**Result**
no graphs at bottom of page

**Expected Result**
graphs shown - as they are for an admin user

**Screenshots**

<img width="1535" alt="2023-09-27_11-28-08" src="https://github.com/rancher/dashboard/assets/89212856/61c348bc-db71-4b15-8ed6-997249c7c8f7">

**Additional context**

**Comment by nwmac** [comment]
The issue here is that the UI tries to fetch the `endpoints` resource to get the endpoints for the links - the user does not have permission to read those.

**Comment by ronhorton** [comment]
![2023-10-02_10-56-54](https://github.com/rancher/dashboard/assets/89212856/2c533589-575f-409f-9bf5-bfdaefaef19e)
![2023-10-02_10-56-31](https://github.com/rancher/dashboard/assets/89212856/c1a35247-a4c6-4b45-930f-14daccd35915)

403's
Does this question require a backport? 403s also happening in 2.7.6.

**Comment by mantis-toboggan-md** [comment]
Blocked by https://github.com/rancher/rancher/issues/43030

**Comment by ronhorton** [comment]
**Pass** verified fixed in rancher `v2.8-01127d844339ea8ab498f334121e2956be610409-head`

1. Enable monitoring v2 on downstream cluster
2. Create a project in this downstream cluster
3. Create a namespace in this project
4. Create an nginx workload in this namespace
5. Open grafana as admin user (from monitoring page) and save the URL for later use
6. Edit the members of the project and add a user with the "Read-only" role
7. Run this in the kubectl shell for the cluster:
8. kubectl create clusterrolebinding view-monitoring-ui --clusterrole=monitoring-ui-view --user=userID
9. Sign in as new user and navigate to downstream cluster page

`Monitoring graphs are at the bottom of the page now.`
<img width="766" alt="2023-10-10_13-25-44" src="https://github.com/rancher/dashboard/assets/89212856/108ae486-02ba-4859-b584-9737b28c5a0b">

**Comment by ronhorton** [comment]
separated 403 errors in a separate ticket (https://github.com/rancher/dashboard/issues/9890)

This ticket's main issue: the graphs at the bottom of the cluster page for a read-only user - is fixed and verified.

**Comment by nwmac** [comment]
@ronhorton Your comment suggest you want this reverted? Is this going to be re-opened? The "QA Review" label is used for QA to review issues where the dev team have covered the test needs through automated test, BTW.

**Comment by ronhorton** [comment]
@nwmac yes, it should be reverted.

I had just updated the zube board, so the status had not synced yet. I'll make sure where i should park something while it's being decided on for next time.

**Comment by prachidamle** [comment]
yes @nwmac @mantis-toboggan-md Thanks for finding a fix for this but we will have to revert it for now to stay true to the documented design. Here is the reasoning for reverting this currently - 

Per the current documentation, a “Read-Only” user (a user without edit permissions) should only be able to view the monitoring links provided externally to them, nothing should be available for them in monitoring pane - and that is as per current documentation. 

https://ranchermanager.docs.rancher.com/integrations-in-rancher/monitoring-and-alerting/rbac-for-monitoring#users-with-rancher-based-permissions

```
A User bound to the View Monitoring Rancher Role only has permissions to access external Monitoring UIs if provided links to those UIs. In order to access the Monitoring Pane to get those links, the User must be a Project Member of at least one Project.
```

Any bugs we log are basically asking for a change in design and should be fixed only after discussing with Product. 
Product has not updated the design/documentation yet and this would be a feature request for the future where we have to analyse the impact of allowing this for a "read-only" user, so we should not push this fix in now.

**Comment by mantis-toboggan-md** [comment]
@prachidamle the UI bug here was that the graphs were hidden for users who couldn't view the monitoring app - is that actually desired behavior? Is there some other resource the UI should be checking to determine access to the embedded graphs? I'm just double-checking because the monitoring app restriction appears to have been an unintentional consequence of fixes related to the local cluster grafana url being different between monitoring versions.

**Comment by geethub97** [comment]
@mantis-toboggan-md The monitoring-ui-view role does not have permissions to get the endpoints, only list them (https://github.com/rancher/charts/blob/706052fd6f93d216c55d6528bc5de3a5cf06d440/charts/rancher-monitoring/103.0.0%2Bup45.31.1/templates/rancher-monitoring/clusterrole.yaml#L134). Is it possible to add another permissions check for the `get` verb?

**Comment by mantis-toboggan-md** [comment]
@geethub97 We can update the UI to check ednpoints permission and hide the graphs if the user doesn't have get permissions on endpoints, does that sound right?

**Comment by mantis-toboggan-md** [comment]
AFACIT the ui can't distinguish between list and get permissions just looking at the schema: either way it includes GET in collection methods and has links.collection defined. Instead I've made a PR to update the cluster page to only show graphs when the user can view the grafana endpoint: this matches how the grafana, alertmanager, and prometheus links are enabled on the monitoring index page.

**Comment by ronhorton** [comment]
**Pass** verified reverted in `rancher v2.8.0-rc3`

Read-only user can no longer see monitoring graphs at bottom of cluster page. 

1. created downstream ec2 server
2. installed rancher-monitoring:103.0.0+up45.31.1
3. created standard user
4. created test project & namespace
5. added standard user as project member with read-only access
6. logged in as new user

`no graphs at bottom of monitoring page`
